main.go

```diff
@@ -0,0 +1,295 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"crypto/ecdsa"
+	"github.com/dgrijalva/jwt-go"
+
+	"github.com/codahale/blake2"
+	"github.com/gomodule/redigo/redis"
+	"github.com/tidwall/resp"
+)
+
+var jwtPubKey *ecdsa.PublicKey
+
+type ZdbBackend struct {
+	Server redis.Conn
+}
+
+type Session struct {
+	Authenticated bool
+	Username      string
+	Backend       *ZdbBackend
+}
+
+type Sessions struct {
+	List    map[*resp.Conn]*Session
+	Backend *ZdbBackend
+}
+
+//
+// its you online
+//
+const (
+	iyoPubKey = `-----BEGIN PUBLIC KEY-----
+MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAES5X8XrfKdx9gYayFITc89wad4usrk0n2
+7MjiGYvqalizeSWTHEpnd7oea9IQ8T5oJjMVH5cc0H5tFSKilFFeh//wngxIyny6
+6+Vq5t5B0V0Ehy01+2ceEon2Y0XDkIKv
+-----END PUBLIC KEY-----`
+)
+
+func cryptoInit() {
+	var err error
+
+	jwtPubKey, err = jwt.ParseECPublicKeyFromPEM([]byte(iyoPubKey))
+	if err != nil {
+		log.Panicf("failed to parse pub key:%v\n", err)
+	}
+}
+
+//
+// list of sessions
+//
+func NewSessions(backend *ZdbBackend) *Sessions {
+	return &Sessions{
+		List:    make(map[*resp.Conn]*Session),
+		Backend: backend,
+	}
+}
+
+// single session
+func (ss *Sessions) NewSession() *Session {
+	return &Session{
+		Authenticated: false,
```

Comment:
False and empty string are the default value for boolean and string, so you don't have to write them explicitly. Not that explicit it bad, just FYI